src/types/mod.rs

```diff
@@ -587,6 +587,16 @@ impl JsObject {
             Handle::new_internal(JsObject(local))
         }
     }
+
+    #[cfg(feature = "napi-8")]
+    pub fn freeze<'a, C: Context<'a>>(self, cx: &mut C) -> bool {
```

Comment:
These methods should return `NeonResult<()>` to reflect that an exception can be thrown.

```
> Object.freeze(Buffer.alloc(1))
Uncaught TypeError: Cannot freeze array buffer views with elements
    at Function.freeze (<anonymous>)
```